sql/system_variables.go

```diff
@@ -0,0 +1,2459 @@
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_cache_size", 0, 16384, true),
+		Default: int64(-1),
+	},
+	"thread_handling": {
+		Name: "thread_handling",
+		Scope: SystemVariableScope_Global,
+		Dynamic: false,
+		SetVarHintApplies: false,
+		Type: NewSystemEnumType("thread_handling", "no-threads", "one-thread-per-connection", "loaded-dynamically"),
+		Default: "one-thread-per-connection",
+	},
+	"thread_pool_algorithm": {
+		Name: "thread_pool_algorithm",
+		Scope: SystemVariableScope_Global,
+		Dynamic: false,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_algorithm", 0, 1, false),
+		Default: int64(0),
+	},
+	"thread_pool_high_priority_connection": {
+		Name: "thread_pool_high_priority_connection",
+		Scope: SystemVariableScope_Both,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_high_priority_connection", 0, 1, false),
+		Default: int64(0),
+	},
+	"thread_pool_max_active_query_threads": {
+		Name: "thread_pool_max_active_query_threads",
+		Scope: SystemVariableScope_Global,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_max_active_query_threads", 0, 512, false),
+		Default: int64(0),
+	},
+	"thread_pool_max_unused_threads": {
+		Name: "thread_pool_max_unused_threads",
+		Scope: SystemVariableScope_Global,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_max_unused_threads", 0, 4096, false),
+		Default: int64(0),
+	},
+	"thread_pool_prio_kickup_timer": {
+		Name: "thread_pool_prio_kickup_timer",
+		Scope: SystemVariableScope_Global,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_prio_kickup_timer", 0, 4294967294, false),
+		Default: int64(1000),
+	},
+	"thread_pool_size": {
+		Name: "thread_pool_size",
+		Scope: SystemVariableScope_Global,
+		Dynamic: false,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_size", 1, 512, false),
+		Default: int64(16),
+	},
+	"thread_pool_stall_limit": {
+		Name: "thread_pool_stall_limit",
+		Scope: SystemVariableScope_Global,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_stall_limit", 4, 600, false),
+		Default: int64(6),
+	},
+	//TODO: implement block sizes
+	//"thread_stack": {
+	//	Name: "thread_stack",
+	//	Scope: SystemVariableScope_Global,
+	//	Dynamic: false,
+	//	SetVarHintApplies: false,
+	//	Type: NewSystemUintType("thread_stack", 131072, 18446744073709551615),
+	//	Default: uint64(286720),
+	//},
+	"time_zone": {
+		Name: "time_zone",
+		Scope: SystemVariableScope_Both,
+		Dynamic: true,
+		SetVarHintApplies: true,
+		Type: NewSystemStringType("time_zone"),
+		Default: "SYSTEM",
+	},
+	//TODO: this needs to be a function
+	"timestamp": {
+		Name: "timestamp",
+		Scope: SystemVariableScope_Session,
+		Dynamic: true,
+		SetVarHintApplies: true,
+		Type: NewSystemDoubleType("timestamp", 1, 2147483647),
+		Default: float64(1000.0),
```

Comment:
Is this what you mean by "needs to be a function"? Might need to special case it, or make a Default method on the new type interface

Reply:
I think timestamp is the only variable with function-like behavior. But a Default method wouldn't work, as I'm not going to have it as "run default expression unless assigned to at some point", but rather just store the value in a big map and stick the default value in that map. Makes changes and whatnot easier.